metapool/tests/test_sample_sheet.py

```diff
@@ -110,14 +110,15 @@ def test_sample_sheet_roundtripping(self):
                 # After parsing, the contents of the written file are the same
                 # because comments and empty lines are ignored in the current
                 # API.
+                old_filename = filename
```

Comment:
The issue is if there's an issue with any of the tests for the three comments/white-space variations for self.good_ss, the user will get an error saying there's a problem with self.good_ss, when it's not actually that file.

Reply:
Then, wouldn't it be easier to move the `if filename in ...` block below the `with open(...` or am I reading that wrongly?

Reply:
The latter. The test loads a sample-sheet into an KLSampleSheet object, writes it back out to file, and then compares the raw lines of the new file against the original. Since some of the sample-sheets are just copies of the good sheet with comments and/or empty lines, they need to be compared against the good-sheet because metapool ignores comments and empty lines. However, if a test fails, we don't want to report there is an issue with good-sheet when observed is actually one of the variants.

Moving the 'with open' above the 'if filename in' in this case would be changing the filename after the test has been performed, which would serve no purpose.

I had to read through the test carefully myself to make sure I had it right so I added some comments and changed a few variable names to make it easier to read. This might help.

Reply:
I think I get it now. It almost sounds like they should be 2 different tests ... but this is now passing and I think it makes sense.